shell/spy.go

```diff
@@ -1,19 +1,32 @@
 package shell
 
 import (
-	"bytes"
+	"fmt"
+	"io"
 	"log"
 	"os/exec"
+	"strings"
 )
 
-// Execute accepts a command argument and returns the result of the executed command
-func Execute(script string) string {
-	cmd := exec.Command(script)
-	var out bytes.Buffer
-	cmd.Stdout = &out
+// Parser represents a function that parses string text
+// and returns a tring and slice of strings
+type Parser func(string) (string, []string)
+
+// ParseCommand accepts a string and returns the command name and args
+func ParseCommand(cmd string) (string, []string) {
+	stringResult := strings.Split(cmd, " ")
+	return stringResult[0], stringResult[1:]
```

Comment:
We're not at home to Mr YOLO! There's no guarantee that `stringResult` has _any_ elements, so if it doesn't, this line will panic. Any time we reference a slice element, we need to make sure that it actually exists first.